custom_metric.py

```
@@ -0,0 +1,329 @@
+from typing import Optional
+from deepeval.metrics import BaseMetric, AnswerRelevancyMetric, FaithfulnessMetric
+from deepeval.test_case import LLMTestCase
+from typing import Optional, List, Union
```

Ruff (0.8.2)

Redefinition of unused `Optional` from line 1

Remove definition: `Optional`

(F811)